src/files/BrsFile.ts

```diff
@@ -121,14 +134,93 @@ export class BrsFile {
         }
     }
 
-
-    public parser: Parser;
+    public parser = new Parser();
 
     public fileContents: string;
 
+    private logDebug(...args) {
+        this.program.logger.debug('BrsFile', chalk.green(this.pkgPath), ...args);
+    }
+
+    /**
+     * If this is a typedef file
+     */
+    public isTypedef: boolean;
+
+    /**
+     * The key to find the typedef file in the program's files map.
+     * A falsey value means this file is ineligable for a typedef
+     */
+    public typedefKey?: string;
+
+    /**
+     * If the file was given type definitions during parse
+     */
+    public get hasTypedef() {
```

Comment:
This feels like a hidden "relatively heavy" operation, and the `cache` approach doesn't feel right. I'd think a more explicit pattern for setting the flag would be preferable.

Reply:
Yeah, good point. I refactored my code accordingly.

Reply:
I thought I needed the evaluation of typedef to be lazy, but after some testing, I verified that the dependency graph events are emitted after the file is finalized, so it _is_ safe to simply resolve these at the time of the event. [Here](https://github.com/rokucommunity/brighterscript/pull/188/files#diff-abb7837cc5611725991e1566d0a4924f98e12e39cecd95dc7f9ccceb6ff658f9R200) and [here](https://github.com/rokucommunity/brighterscript/pull/188/files#diff-abb7837cc5611725991e1566d0a4924f98e12e39cecd95dc7f9ccceb6ff658f9R59) are the relevant locations where I resolve this information now.